src/portpicker.py

```diff
@@ -142,7 +142,7 @@ def pick_unused_port(pid=None):
         the current process's PID is used.
 
     Returns:
-      A port number that is unused on both TCP and UDP.
+      A port number that is unused on both TCP and UDP. None on error.
```

Comment:
You're right. I was trying to make pick_unused_port's docstring match what _pick_unused_port_without_server has *said* it would do, but in reality, it never did, since it was spinning forever.

I'll make this raise an exception, instead, and update both docstrings.

Reply:
well, I guess we do have a couple `return None`s in the `get_port_from_port_server()` implementation which ultimately result in a return value of None in some cases... so maybe just returning None isn't wholly unexpected.

My intiution is that users do not expect to see a `None` return value though.  So turning all None returns into an exception _seems_ like a good idea.

Historically I suspect this was done this way due to the `main()` use case as a shell tool (people do use portpicker this way quite often).  The error message is printed to stderr, and we exit 1.

Reply:
On the other hand, when using the module's advertised API, having `get_port_from_port_server` return `None` allows `pick_unused_port` to gracefully fall through to `_pick_unused_port_without_server`.

If we change `get_port_from_port_server` to raise an exception instead of returning `None`, I suspect we'd just catch that error in `pick_unused_port` to maintain the fall-back-to-random-ports approach. Then we'd be changing the `get_port_from_port_server` API just for the sake of consistency. While that function's docstring says it's not meant for external use, it *is* part of the module's exported functions, so there could be callers that depend on it returning None. (So I guess it would necessitate a version bump.)

Happy to go either way with it; up to you.